cherab/jet/spectroscopy/kt3/kt3.py

```diff
@@ -0,0 +1,405 @@
+
+    The method uses the FibreOpticGroup to provide a set of sightlines approximating the KT3A/KT3B diagnostics. Sightlines are
+    approximated with SpectroscopicFibreOptic (FibreOptic observers).
+    
+    :param spectrometer: Spectrometer name
+    :param origin: A 1D array with [R, Z] cylindrical coordinates of the KT3 origin in the tokamak frame, which is the position
+      of origin of the FibreOpticGroup.
+    :param spots_rz: A 2xN array of [[R0, ...], [Z0, ...]] cylindrical coordinates in the tokaak frame of the spot centers on the tile-5 plane (divertor).
+    :param min_wavelength: Minimum observed wavelength
+    :param max_wavelength: Maximum observed wavelength
+    :param spectral_bins: Number of spectral bins
+    :param pixel_samples: Number of rays to be used in observation for every sight-line (default 1e3).
+    :param toroidal_angle: The toroidal position of the diagnostic (default 303.75 degrees)
+    :param parent: parent node of the returned FibreOpticGroup
+
+    :return: FibreOpticGroup instance with sight-lines approximating KT3 diagnostic
+    """
+
+    # The spots are supposed to be "touching" the neighbout spots
+    spot_radius = _calculate_spot_radii(spots_rz)
+    acceptance_angles = _calculate_acceptance_angles(origin_rz, spots_rz, spot_radius)
+
+    # the group frame is only rotated world (tokamak frame), so the transform of sightlines in
+    # the group is performed in a single poloidal plane
+    group = FibreOpticGroup()
+    group.name = spectrometer
+    group.parent = parent
+    group.transform = rotate_z(toroidal_angle)
+        
+    origin = []
+    for acceptance_angle, spot_rz in zip(acceptance_angles, spots_rz.T):
+        # direction in the RZ group poloidal plane
+        direction = Vector3D(spot_rz[0] - origin_rz[0], 0, spot_rz[1] - origin_rz[1])
+
+        # Position of the observers in the RZ group poloidal plane
+        # shift origin by a mm and calculate the fibre radius from the acceptance angle
+        origin = Point3D(origin_rz[0], 0, origin_rz[1]) + (1e-3 * direction.normalise())
+        fibre_radius = 1e-3 * tan(radians(acceptance_angle))
+
+        sight_line = SpectroscopicFibreOptic(origin, direction, radius=fibre_radius, acceptance_angle=acceptance_angle) 
+        sight_line.min_wavelength = min_wavelength
+        sight_line.max_wavelength = max_wavelength
+        sight_line.spectral_bins = spectral_bins
+        sight_line.pixel_samples = pixel_samples
+
+        sight_line.connect_pipelines([(SpectralRadiancePipeline0D, "radiance r:{:4.3f}".format(spot_rz[0]), None),
+                                    (SpectralPowerPipeline0D, "power r:{:4.3f}".format(spot_rz[0]), None)])
+        group.add_sight_line(sight_line)
+
+    return group
+
+def kt3_from_sal(spectrometer, pulse_number,min_wavelength=None, max_wavelength=None, spectral_bins=None,
+                 spot_position_correction=True, pulse_time=None, time_selection="nearest",
```

Comment:
I would change the default value for `spot_position_correction` to `False`. The private PPF required for the spot correction may not be available.